src/dsp_tools/models/connection.py

```diff
@@ -27,240 +29,260 @@ def check_for_api_error(response: requests.Response) -> None:
-            with open(filename, "w", encoding="utf-8") as f:
-                json.dump(logobj, f, indent=4)
+        headers = {}
+        if jsondata:
+            headers = {"Content-Type": "application/json; charset=UTF-8"}
+        if self.token:
+            headers["Authorization"] = "Bearer " + self.token
+
+        response = requests.post(
+            self.server + path,
+            headers=headers,
+            data=jsondata,
+            timeout=timeout,
+        )
+        self.write_log_file(
+            method="POST",
+            route=path,
+            headers=headers,
+            jsondata=jsondata,
+            params=None,
+            response=response,
+        )
         check_for_api_error(response)
-        result = response.json()
-        return result
+        json_response = cast(dict[str, Any], response.json())
+        return json_response
 
-    def get(self, path: str, headers: Optional[dict[str, str]] = None) -> dict[str, Any]:
-        """
-        Get data from a server using a HTTP GET request
-        :param path: Path of RESTful route
-        :param headers: ...
-        :return: Response from server
+    def get(
+        self,
+        path: str,
+        headers: Optional[dict[str, str]] = None,
+    ) -> dict[str, Any]:
         """
+        Make a HTTP GET request to the server to which this connection has been established.
 
-        if path[0] != "/":
-            path = "/" + path
-        if not self._token:
-            if not headers:
-                response = requests.get(self._server + path, timeout=20)
-            else:
-                response = requests.get(self._server + path, headers, timeout=20)
-        else:
-            if not headers:
-                response = requests.get(
-                    self._server + path,
-                    headers={"Authorization": "Bearer " + self._token},
-                    timeout=20,
-                )
-            else:
-                headers["Authorization"] = "Bearer " + self._token
-                response = requests.get(self._server + path, headers, timeout=20)
+        Args:
+            path: Path of RESTful route
+            headers: headers for the HTTP request
 
+        Returns:
+            response from server
+        """
+        if not path.startswith("/"):
+            path = "/" + path
+        if not headers:
+            headers = {}
+        if self.token:
+            headers["Authorization"] = "Bearer " + self.token
+
+        response = requests.get(url=self.server + path, headers=headers, timeout=20)
+        self.write_log_file(
+            method="GET",
+            route=path,
+            headers=headers,
+            jsondata=None,
+            params=None,
+            response=response,
+        )
         check_for_api_error(response)
-        json_response = response.json()
+        json_response = cast(dict[str, Any], response.json())
         return json_response
```

Comment:
maybe I'm going a bit over board now... but this is almost identical for all HTTP methods; could the shared logic be extracted to a separate method?